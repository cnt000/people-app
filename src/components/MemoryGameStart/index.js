import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Div = styled.div`
  margin: auto auto;
  text-align: center;
`

const MemoryGameStart = ({ onClick }) => {
  return (
    <Div onClick={onClick}>
      <h1>Memory Game</h1>
      <p>Click to start</p>
    </Div>
  )
}

MemoryGameStart.propTypes = {
  onClick: PropTypes.func.isRequired
}

export default MemoryGameStart