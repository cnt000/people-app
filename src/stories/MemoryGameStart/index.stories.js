import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import '@storybook/addon-viewport/register'

import MemoryGameStart from "../../components/MemoryGameStart"

storiesOf('MemoryGameStart', module)
  .add('default', () => <MemoryGameStart onClick={action('clicked')} />)
