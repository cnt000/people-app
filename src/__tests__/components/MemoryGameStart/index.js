import React from 'react';
import ReactDOM from 'react-dom';
import MemoryGameStart from '../../../components/MemoryGameStart';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('MemoryGameStart', () => {
  it('should render (without snapshot)', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryGameStart onClick={()=>console.log('click')} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render correctly with snapshot', () => {
    const output = shallow(
      <MemoryGameStart onClick={()=>console.log('click')} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});