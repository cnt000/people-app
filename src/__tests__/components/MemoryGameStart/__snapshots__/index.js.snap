// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoryGameStart should render correctly with snapshot 1`] = `
<styled.div
  onClick={[Function]}
>
  <h1>
    Memory Game
  </h1>
  <p>
    Click to start
  </p>
</styled.div>
`;
